eventhubs/data-plane/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/MessageSender.java

```diff
@@ -417,6 +417,11 @@ public void onEvent() {
                 } else if (completionException instanceof EventHubException
                         && !((EventHubException) completionException).getIsTransient()) {
                     this.cancelOpen(completionException);
+                } else {
```

Comment:
At this point it could be some other type of exception, or it could be a transient EventHubException. Transient EventHubExceptions should drop through without a log, but if it's something else we want to log that and do a cancelOpen().

Reply:
Ahh, I see what you mean. Thanks for the feedback. :) I've updated the PR. How does it look now?

Reply:
Looks good! Breaking it into nested ifs like this makes the code easier to read, too.